Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/AnswersController.cs

```diff
@@ -136,23 +138,24 @@ public async Task<ActionResult<AnswerViewModel>> DeleteAnswer(int id)
             if (user == null)
             {
                 return Unauthorized();
-            }
+            }
+
+            var roles = await userManager.GetRolesAsync(user);
 
             var answer = await _context.Answers.FindAsync(id);
             if (answer == null)
             {
                 return NotFound();
             }
-            if (answer.UserID != user.Id)
+
+            if (answer.UserID != user.Id && roles[0] != "Admin")
```

Comment:
If you use [Authorize(Roles = "Admin")] to [Authorize] you would not need an if statement to check if the user is an admin.

Reply:
The if statement is to modify the return so that you can delete a question without being redirected because you are not the original poster as An admin